packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1373,6 +1438,47 @@ function generateExportName(stackExports: Construct, id: string) {
   return prefix + localPart.slice(Math.max(0, localPart.length - maxLength + prefix.length));
 }
 
+function isExportAList(exportedValue: any, resolved: any) {
+  const target = exportedValue.target;
+  const resolvedGetAtt = resolved['Fn::GetAtt'];
+  const resolvedRef = (resolved['Fn::Ref'] ?? resolved.Ref);
+
+  if (resolvedGetAtt && resolvedRef) {
+    throw new Error(`found a reference with both 'Fn::GetAtt' and 'Fn::Ref' ${resolved}`);
+  }
+
+  if (resolvedGetAtt) {
```

Comment:
This whole process seems crazy dangerous to me. I'm also not seeing any unit tests that would test the assumptions on our generated L1 code, which this seems to rely on.

Can we not do `Array.isArray(exportedValue)` and be close enough? Bonus: customers get some control over it in case we get it wrong.

If nothing else, I'd rather add "type hinting" as a feature to tokens, rather than do this introspection into guessing where it came from.